### PR TITLE
mkdir: fail if no arguments are provided

### PR DESCRIPTION
**Pull request by andrewliebenow**

### CODE REVIEW

**Comment by andrewliebenow** [comment]
https://github.com/uutils/coreutils/pull/6790 can be used instead of this for the no arguments fix, but we still need the fix in this PR for:

```
❯ /usr/bin/mkdir '' ; printf '%s\n' "${?:?}" 
/usr/bin/mkdir: cannot create directory ‘’: No such file or directory
1

❯ coreutils mkdir '' ; printf '%s\n' "${?:?}"
0
```

I'll update this PR accordingly when #6790 is merged.

**Review by BenWiederhake — COMMENTED** [review]
I like this PR (and prefer it over #5790) because it emits a clearer error message, and also handles the case of emptystring-arguments.

However, the 10 CI failures make it unclear whether this breaks anything. Can you update to current main? This would include #6972, which should resolve some (all?) CI failures.

File: src/uu/mkdir/src/mkdir.rs [hunk 1]
```diff
@@ -159,6 +176,13 @@ fn exec(dirs: ValuesRef<OsString>, recursive: bool, mode: u32, verbose: bool) ->
 /// To match the GNU behavior, a path with the last directory being a single dot
 /// (like `some/path/to/.`) is created (with the dot stripped).
 pub fn mkdir(path: &Path, recursive: bool, mode: u32, verbose: bool) -> UResult<()> {
+    if path.as_os_str().is_empty() {
+        return Err(USimpleError::new(
+            1,
+            "cannot create directory ‘’: No such file or directory".to_owned(),
```
[the remark this answers is not in the document]
Comment (andrewliebenow) [reply src/uu/mkdir/src/mkdir.rs:182]: I have not been using LC_ALL=C when checking GNU Core Utilities' behavior. There are some other PRs I've made that probably need to be updated now that I know about this.

**Comment by andrewliebenow** [comment]
I had missed a "." in the test, fixed now

**Comment by sylvestre** [comment]
hmm, i am sorry, i took this PR: https://github.com/uutils/coreutils/pull/6790
as it seems to do the same thing but shorter

**Comment by andrewliebenow** [comment]
> #6790 can be used instead of this for the no arguments fix, but we still need the fix in this PR for:
> 
> ```
> ❯ /usr/bin/mkdir '' ; printf '%s\n' "${?:?}" 
> /usr/bin/mkdir: cannot create directory ‘’: No such file or directory
> 1
> 
> ❯ coreutils mkdir '' ; printf '%s\n' "${?:?}"
> 0
> ```
> 
> I'll update this PR accordingly when #6790 is merged.

Going to just make a new PR instead

**Pull request closed without merging**